packages/nodes-base/nodes/MessageBird/MessageBird.node.ts

```diff
@@ -0,0 +1,354 @@
+import { IExecuteFunctions } from 'n8n-core';
+import {
+	IDataObject,
+	INodeTypeDescription,
+	INodeExecutionData,
+	INodeType
+} from 'n8n-workflow';
+
+import { messageBirdApiRequest } from './GenericFunctions';
+
+export class MessageBird implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'MessageBird',
+		name: 'messageBird',
+		icon: 'file:messagebird.png',
+		group: ['transform'],
+		version: 1,
+		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+		description: 'Sending SMS',
+		defaults: {
+			name: 'MessageBird',
+			color: '#2481d7'
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'messageBirdApi',
+				required: true
+			}
+		],
+		properties: [
+			{
+				displayName: 'Resource',
+				name: 'resource',
+				type: 'options',
+				options: [
+					{
+						name: 'SMS',
+						value: 'sms'
+					}
+				],
+				default: 'sms',
+				description: 'The resource to operate on.'
+			},
+
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				displayOptions: {
+					show: {
+						resource: ['sms']
+					}
+				},
+				options: [
+					{
+						name: 'Send',
+						value: 'send',
+						description: 'Send text messages (SMS)'
+					}
+				],
+				default: 'send',
+				description: 'The operation to perform.'
+			},
+
+			// ----------------------------------
+			//         sms:send
+			// ----------------------------------
+			{
+				displayName: 'From',
+				name: 'originator',
+				type: 'string',
+				default: '',
+				placeholder: '14155238886',
+				required: true,
+				displayOptions: {
+					show: {
+						operation: ['send'],
+						resource: ['sms']
+					}
+				},
+				description: 'The number from which to send the message'
+			},
+			{
+				displayName: 'To',
+				name: 'recipients',
+				type: 'string',
+				default: '',
+				placeholder: '14155238886/+14155238886',
+				required: true,
+				displayOptions: {
+					show: {
+						operation: ['send'],
+						resource: ['sms']
+					}
+				},
+				description: 'all recipients separated by commas'
+			},
+
+			{
+				displayName: 'Message',
+				name: 'message',
+				type: 'string',
+				default: '',
+				required: true,
+				displayOptions: {
+					show: {
+						operation: ['send'],
+						resource: ['sms']
+					}
+				},
+				description: 'The message to be send'
+			},
+			{
+				displayName: 'Additional Fields',
+				name: 'additionalFields',
+				type: 'collection',
+				placeholder: 'Add Fields',
+				default: {},
+				options: [
+					//date-time format
+					{
+						displayName: 'Created Date-time',
+						name: 'createdDatetime',
+						type: 'dateTime',
+						placeholder: '2011-08-30T09:30:16.768-04:00',
+						default: '',
+						description:
+							'The date and time of the creation of the message in RFC3339 format (Y-m-dTH:i:sP).'
```

Comment:
These should be inline unless it's a styleguide thing. I need to build a node myself so I get what's going on here :).